src/lib/radon/artifacts.ts

```diff
@@ -40,9 +57,60 @@ class Class {
 }
 
 export class RadonRequest extends Class {   
-    
-    public static from(hexString: string) {
-        return Utils.decodeRadonRequest(hexString)
+   
+    /**
+     * Decodes a RadonRequest artifact out from a Protobuf-serialized bytecode.
+     * @param bytecode Radon request bytecode.
+     * @returns RadonRequest object.
+     */
+    public static fromBytecode(bytecode: any) {
+        let buffer
+        if (isHexString(bytecode)) {
+            buffer = fromHexString(bytecode)
+        } else if (bytecode instanceof Uint8Array) {
+            buffer = Buffer.from(bytecode)
+        } else if (bytecode instanceof Buffer) {
+            buffer = bytecode
+        } else {
+            throw new TypeError(`RadonRequest: unsupported bytecode format: ${bytecode}`)
+        }
+        const obj: any = RADRequest.decode(buffer)
+        const retrieve = obj.retrieve.map((retrieval: any) => {
+            const specs: any = {}
+            if (retrieval?.url) { specs.url = retrieval.url }
+            if (retrieval?.headers) {
+                specs.headers = retrieval.headers.map((stringPair: any) => [
+                    stringPair.left,
+                    stringPair.right
+                ])
+            }
+            if (retrieval?.body && retrieval.body.length > 0) {
+                specs.body = utf8ArrayToStr(Object.values(retrieval.body))
+            }
+            // if (retrieval?.script) specs.script = decodeRadonScript(toHexString(retrieval.script))
```

Comment:
```suggestion
```

Reply:
Actually, it needs to be uncomented, not deleted.